tests/test_api.py

```diff
@@ -15,76 +16,127 @@ def test_defaults():
 @pytest.mark.skip(reason="file format not implemented yet")
 def test_kml_extract_bbox():
     assert geoextent.fromFile("tests/testdata/kml/aasee.kml", bbox=True) == [7.594213, 51.942466, 7.618246, 51.957278]
 
+
 @pytest.mark.skip(reason="file format not implemented yet")
 def test_gpkg_extract_bbox():
     assert geoextent.fromFile("tests/testdata/nc/nc.gpkg", bbox=True) == [33.882, -84.3239, 36.5896, -75.457]
 
+
 @pytest.mark.skip(reason="file format not implemented yet")
 def test_gml_extract_bbox():
-    assert geoextent.fromFile("tests/testdata/gml/clc_1000_PT.gml", bbox=True) == [-17.5420724159224, 32.3966928193202, -6.95938792923511, 39.3011352746141]
+    assert geoextent.fromFile("tests/testdata/gml/clc_1000_PT.gml", bbox=True) == [-17.5420724159224, 32.3966928193202,
+                                                                                   -6.95938792923511, 39.3011352746141]
+
+def test_empty_folder():
+    os.mkdir('tests/testdata/folders/temp_empty_folder')
+    result = geoextent.fromDirectory('tests/testdata/folders/temp_empty_folder', bbox=True, tbox=True)
+    assert "bbox" not in result
+    assert "tbox" not in result
+    os.rmdir('tests/testdata/folders/temp_empty_folder')
+
+def test_folder_one_file():
+    result = geoextent.fromDirectory('tests/testdata/folders/folder_one_file', bbox=True, tbox=True)
+    assert "bbox" in result
+    assert "tbox" in result
+    assert result["bbox"] == [7.60168075561523, 51.9488147720619, 7.64725685119629, 51.9746240298775]
+    assert result["tbox"] == ['2018-11-14', '2018-11-14']
 
-@pytest.mark.skip(reason="file format not implemented yet")
 def test_folder_multiple_files():
-    assert [7.6016807556152335, 51.94881477206191, 7.647256851196289, 51.974624029877454] in geoextent.fromFile('tests/testdata/folder/', bbox=True)
-    assert [6.574722, 51.434444, 4.3175, 53.217222] in geoextent.fromFile('tests/testdata/folder', bbox=True)
-    assert [292063.81225905, 5618144.09259115, 302531.3161606, 5631223.82854667] in geoextent.fromFile('tests/testdata/folder', bbox=True)
+    result = geoextent.fromDirectory('tests/testdata/folders/folder_two_files', bbox=True, tbox=True)
+    assert "bbox" in result
+    assert "tbox" in result
+    assert result["bbox"] == [2.05233338763921, 41.3170385224048, 7.64725685119629, 51.9746240298775]
+    assert result["tbox"] == ['2018-11-14', '2019-09-11']
 
+def test_folder_nested_files():
+    result = geoextent.fromDirectory('tests/testdata/folders/nested_folder', bbox=True, tbox=True)
+    assert "bbox" in result
+    assert "tbox" in result
+    assert result["bbox"] == [7.60168075561523, 34.7, 142.0, 51.9746240298775]
+    assert result["tbox"] == ['2017-04-08', '2020-02-06']
+
+def test_empty_zipfile():
+    folder_name = "tests/testdata/folders/temp_empty_folder"
+    os.mkdir(folder_name)
+    hf.create_zip(folder_name)
+    result = geoextent.fromDirectory(folder_name+'.zip', bbox=True, tbox=True)
+    os.remove(folder_name + '.zip')
```

Comment:
If there is an error in the line before, the `os.remove(..)` will not be called and the test will leave a file artefact.

Please take advantage of the `with ...` statement to ensure the files created for the tests are cleaned up, see https://docs.python.org/3.4/library/tempfile.html#examples

Maybe something like

```python
with tempfile.TemporaryFile(suffix = 'zip') as zipfile:
    create_zip(zipfile, folder_name)
    result = geoextent.fromDirectory(zipfile, bbox=True, tbox=True)
    assert "bbox" not in result
    assert "tbox" not in result
```

with a `create_zip()` function that also takes a target file name - and that's it.